src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -636,4 +672,19 @@ public static void checkCompleted(CompletionStage<?> stage) {
     }
   }
 
+  static <K, V> Map<K, V> zip(final List<K> keys, final List<V> values) {
```

Comment:
I would probably have kept this as a private implementation detail - except for the list sizes, the rest of the logic gets tested from the outside via `joinMap`. And that should never be possible given the implementation of `joinMap`.

Reply:
I think zip could be a useful function on its own.
Random thoughts:
1) Zip could be public
2) Rename to zipAsMap
3) Inputs could be Iterable instead of list (you'd have to skip the size check and fail if they don't end at the same time)

Reply:
Making it a part of the public API can also be a good choice..

Reply:
Right. But then again, it didn't really strike me as a completable-futures thing. How about I make it private for now, and remove the list size check and the tests?